src/lib.rs

```diff
@@ -121,7 +120,7 @@ fn iter_ids<F>(
 where
     F: FnMut(MantaObject, u32) -> Result<(), Error>,
 {
-    let mut mclient = MorayClient::from_str(moray_socket.as_str(), log, None).unwrap();
+    let mut mclient = MorayClient::from_str(moray_socket, log, None).unwrap();
```

Comment:
What happens if from_str panics here? Should we handle it?

Reply:
I changed this one, but this really needs to be fixed throughout.  I filed MANTA-4494.